### PR TITLE
improvements to geet_commonvoice_data script

### PR DESCRIPTION
**Pull request by bmwshop**
# What does this PR do ?

compatibility with MCV versions 2 through8

**Collection**: ASR

# Changelog 
- If tsv file does not include .mp3 extension for files, it is added
- If an mp3 file is empty, it is correctly ignored
- if the archive file is not in the {language}.tar.gz format, it is downloaded in that format anyways for correct downstream processing
- if the archive does not have layered directories (e.g. /version/language/files.tsv), it is correctly processed anyways

# Usage

```python
python get_commonvoice_data.py --data_root /data/temp --log --num_workers=16 --language es -- version cv-corpus-6.1-2020-12-11
python get_commonvoice_data.py --data_root /data/temp --log --num_workers=16 --language es --version cv-corpus-4-2019-12-10
python get_commonvoice_data.py --data_root /data/temp --log --num_workers=16 --language es --version cv-corpus-3
python get_commonvoice_data.py --data_root /data/temp --log --num_workers=16 --language es --version cv-corpus-2
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: scripts/dataset_processing/get_commonvoice_data.py [hunk 1]
```diff
@@ -88,9 +88,11 @@ def create_manifest(data: List[tuple], output_name: str, manifest_path: str):
 
     with output_file.open(mode='w') as f:
         for wav_path, duration, text in tqdm(data, total=len(data)):
-            f.write(
-                json.dumps({'audio_filepath': os.path.abspath(wav_path), "duration": duration, 'text': text}) + '\n'
-            )
+            if wav_path != '':
+                # skip invalid input files that could not be converted
```
Comment (titu1994) [inline scripts/dataset_processing/get_commonvoice_data.py:92]: If skipping, log that file xyz could not be passed / added to manifest.

File: scripts/dataset_processing/get_commonvoice_data.py [hunk 2]
```diff
@@ -144,14 +156,21 @@ def main():
         logging.info('Find existing folder {}'.format(target_unpacked_dir))
     else:
         logging.info("Could not find Common Voice, Downloading corpus...")
-
+        
+        # some dataset versions are packaged in different named files, so forcing
+        output_archive_filename = args.language + '.tar.gz'
+        output_archive_filename = os.path.join(data_root, output_archive_filename)
+
+#             '-P',
```
Comment (titu1994) [inline scripts/dataset_processing/get_commonvoice_data.py:164]: Remove comments

**Review by titu1994 — DISMISSED** [review]
Minor comments. Merge when ready

File: scripts/dataset_processing/get_commonvoice_data.py (hunk 1 above)
Comment (bmwshop) [reply scripts/dataset_processing/get_commonvoice_data.py:92]: that is done.

File: scripts/dataset_processing/get_commonvoice_data.py (hunk 2 above)
Comment (bmwshop) [reply scripts/dataset_processing/get_commonvoice_data.py]: done

**Review by titu1994 — APPROVED** [review]
Lgtm

**Pull request merged**